src/Integration/MefServices/VsSessionHost.cs

```diff
@@ -308,9 +308,7 @@ private void RegisterLocalServices()
                 var solution = this.GetService<SVsSolution, IVsSolution>();
                 var store = new SecretStore(SolutionBindingSerializer.StoreNamespace);
                 var legacySerializer = new SolutionBindingSerializer(this);
-                // The SCC wrapper maintains a queue of file writes. For the new provider we want to the file write to be
-                // executed immediately, so we want our own SCC wrapper rather than sharing the one used by the legacy mode.
```

Comment:
The new connected mode now writes multiple files to disk (shared ruleset, config file, and the project-level ruleset files) so there is no reason for it not to use the same ISourceControlledFileSystem to queue the changes.